### PR TITLE
*: refactor the optional arguments for `table.AddRecord` and `index.Create`

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Better code

### What is changed and how it works?

Our current usage of the optional arguments is quite ugly.
Here is a blog post about how to write clean code for this pattern:
https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

### Check List 

Tests 

 - No code

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @jackysp @lysu @crazycs520

File: table/index.go [hunk 1]
```diff
@@ -33,12 +33,25 @@ type CreateIdxOpt struct {
 	SkipCheck       bool // If true, skip all the unique indices constraint check.
 }
 
+// CreateIdxOption is defined for the Create() method of Index interface.
+type CreateIdxOption func(*CreateIdxOpt)
```
Comment (crazycs520) [inline table/index.go]: How about `CreateIdxOptFunc`?
It's hard to distinguish any difference between `CreateIdxOption` and `CreateIdxOpt `.

Comment (tiancaiamao) [reply table/index.go]: Done

**Pull request merged**